### PR TITLE
sharded-test-server: set shard external URL to front-proxy

### PR DESCRIPTION
**Pull request by sttts**
Users with kubectl should keep talking to the front-proxy. The shard external URL is what ends up in `ClusterWorkspace.status.baseURL`. This PR wires the front-proxy into the external URL of the shard.

### CODE REVIEW

File: cmd/sharded-test-server/shard.go [hunk 1]
```diff
@@ -76,6 +76,7 @@ func startShard(ctx context.Context, n int, args []string, servingCA *crypto.CA,
 		"--service-account-key-file=.kcp/service-account.crt",
 		"--service-account-private-key-file=.kcp/service-account.key",
 		"--audit-log-path", auditFilePath,
+		fmt.Sprintf("--shard-external-url=https://%s:%d", hostIP, 6443),
```
Comment (p0lyn0mial) [inline cmd/sharded-test-server/shard.go:79]: okay, this points to the front porxy

**Comment by sttts** [comment]
/override ci/prow/e2e

This does not touch e2e job.

**Pull request merged**